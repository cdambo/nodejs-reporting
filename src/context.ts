export default class Context {
  private readonly globalTags: object;

  public constructor(globalTags: object = {}) {
    this.globalTags = globalTags;
  }

  public mergeTags(tags: object = {}): object {
    return Object.assign({}, this.globalTags, tags);
  }
}
