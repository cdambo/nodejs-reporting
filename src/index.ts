import Context from "./context";

export * from "./metrics";
export { Context };
