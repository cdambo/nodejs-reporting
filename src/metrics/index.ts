export * from "./formats";
export * from "./reporters";
export * from "./integrations";
