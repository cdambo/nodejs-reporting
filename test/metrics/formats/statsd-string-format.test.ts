import { StatsDStringFormat } from "../../../src";

describe("StatsDStringFormat", (): void => {
  describe("format", (): void => {
    it("Formats the metric to an object", (): void => {
      expect(
        StatsDStringFormat().format({
          metric: "increment",
          stat: "metric_name",
          tags: { a: 1, b: 2 }
        })
      ).toMatchSnapshot();
    });

    it("Formats the metric with options", (): void => {
      expect(
        StatsDStringFormat({
          capitalizeMetric: true,
          colors: true,
          compact: true,
          breakLength: 100,
          depth: 2
        }).format({
          metric: "increment",
          stat: "metric_name",
          tags: { a: 1, b: 2, c: { d: 5 } }
        })
      ).toMatchSnapshot();
    });

    it("Formats the tags to snake case keys", (): void => {
      expect(
        StatsDStringFormat().format({
          metric: "increment",
          stat: "metric_name",
          tags: {
            camelCaseTag1: "tagOne",
            camelCaseTag2: "tag_two",
            anotherTag: 5
          }
        })
      ).toMatchSnapshot();
    });

    it("Removes leading non-letters from metric name", (): void => {
      expect(
        StatsDStringFormat().format({
          metric: "increment",
          stat: "123._-/*()#$@!a_real_metric"
        })
      ).toMatchSnapshot();
    });

    it("Replaces with an underscore characters that are not ASCII alphanumerics, underscores, or periods in metric name", (): void => {
      expect(
        StatsDStringFormat().format({
          metric: "increment",
          stat: "a-real_Metric.Name_with+bad#chars!@-#$%^&*()9"
        })
      ).toMatchSnapshot();
    });
  });
});
