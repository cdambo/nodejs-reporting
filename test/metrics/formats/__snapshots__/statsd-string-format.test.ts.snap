// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsDStringFormat format Formats the metric to an object 1`] = `
"{
  increment: {
    stat: 'metric_name',
    tags: {
      a: 1,
      b: 2
    }
  }
}"
`;

exports[`StatsDStringFormat format Formats the metric with options 1`] = `"{ INCREMENT: { stat: [32m'metric_name'[39m, tags: { a: [33m1[39m, b: [33m2[39m, c: [36m[Object][39m } } }"`;

exports[`StatsDStringFormat format Formats the tags to snake case keys 1`] = `
"{
  increment: {
    stat: 'metric_name',
    tags: {
      camel_case_tag_1: 'tagOne',
      camel_case_tag_2: 'tag_two',
      another_tag: 5
    }
  }
}"
`;

exports[`StatsDStringFormat format Removes leading non-letters from metric name 1`] = `
"{
  increment: {
    stat: 'a_real_metric',
    tags: {}
  }
}"
`;

exports[`StatsDStringFormat format Replaces with an underscore characters that are not ASCII alphanumerics, underscores, or periods in metric name 1`] = `
"{
  increment: {
    stat: 'a_real_Metric.Name_with_bad_chars___________9',
    tags: {}
  }
}"
`;
